manifests/charts/gateway/templates/deployment.yaml

```diff
@@ -25,7 +25,7 @@ spec:
         {{- with .Values.revision }}
         istio.io/rev: {{ . }}
         {{- end }}
-        {{- include "gateway.selectorLabels" . | nindent 8 }}
+        {{- include "gateway.labels" . | nindent 8 }}
```

Comment:
Gateway service is using `selectorLabels` in its selector to selector the gateway pods.

Reply:
the `gateway.labels` template includes the `gateway.selectorLabels` (here https://github.com/istio/istio/blob/master/manifests/charts/gateway/templates/_helpers.tpl#L18) so both the service selector and and the selector `matchLabels` in the deployment should be all good. It is just a matter of the pod, which the deployment instantiated through the replica set, also gets the full set of labels as the deployment resource itself.